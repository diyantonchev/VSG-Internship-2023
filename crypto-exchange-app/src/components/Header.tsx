import { useSelector } from 'react-redux';
import { Navbar, Container } from 'react-bootstrap';

import { ApplicationState, User } from '../types';

const Header = () => {
  const user = useSelector<ApplicationState, User>(state => state.user);

  return (
    <Navbar bg='primary' variant='dark'>
      <Container>
        <Navbar.Brand href='/'>Crypto Exchange</Navbar.Brand>
        <Navbar.Collapse className="justify-content-end">
            <Navbar.Text>
              Signed in as: <a role='button'>{user.name}</a>
            </Navbar.Text>
          </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Header;
