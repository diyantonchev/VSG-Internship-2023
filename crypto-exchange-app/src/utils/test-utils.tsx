import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import store from '../redux/store';

type ProvidersProps = {
  children: ReactElement;
};

const AllTheProviders = ({ children } : ProvidersProps) => {
  return (
    <Provider store={store}>
      {children}
    </Provider>
  )
}

const customRender = (ui: ReactElement, options?: object) => (
  render(ui, { wrapper: AllTheProviders, ...options })
);

export * from '@testing-library/react'

// override render method
export { customRender as render }