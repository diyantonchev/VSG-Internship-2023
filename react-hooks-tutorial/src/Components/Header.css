.header-nav ul {
  display: flex;
  padding: 0;
  list-style: none;
}

.header-nav ul li {
  margin-right: 16px;
}