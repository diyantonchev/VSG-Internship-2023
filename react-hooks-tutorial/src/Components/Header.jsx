import { NavLink } from 'react-router-dom';
import './Header.css';

const Header = () => (
  <header>
    <nav className='header-nav'>
      <ul>
        <li>
          <NavLink
            to='/'
          >
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to='/old'>
            The old way
          </NavLink>
        </li>
        <li>
          <NavLink to='/new'>
            The hooks way
          </NavLink>
        </li>
        <li>
          <NavLink to='/useState'>
            useState
          </NavLink>
        </li>
        <li>
          <NavLink to='/useEffect'>
            useEffect
          </NavLink>
        </li>
        <li>
          <NavLink to='/useRef'>
            useRef
          </NavLink>
        </li>
        <li>
          <NavLink to='/useMemo'>
            useMemo
          </NavLink>
        </li>
        <li>
          <NavLink to='/useCallback'>
            useCallback
          </NavLink>
        </li>
        <li>
          <NavLink to='/custom'>
            Custom Hooks
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
