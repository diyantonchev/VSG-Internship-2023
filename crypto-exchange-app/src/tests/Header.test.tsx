
import { render, screen } from 'test-utils';
import Header from '../components/Header';

describe('Header', () => { 
  it('should renders Header component', () => {
    render(<Header />);

    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(screen.getByText('Crypto Exchange')).toBeInTheDocument();
  });

  it('it should match the snapshot', async () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
