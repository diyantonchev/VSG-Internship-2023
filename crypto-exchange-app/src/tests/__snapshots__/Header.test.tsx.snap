// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it should match the snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-primary"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Crypto Exchange
      </a>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <span
          class="navbar-text"
        >
          Signed in as: 
          <a
            role="button"
          />
        </span>
      </div>
    </div>
  </nav>
</div>
`;
